public/pages/Overview/containers/AnomalyDetectionOverview.tsx

```diff
@@ -0,0 +1,376 @@
+  EuiLoadingSpinner,
+} from '@elastic/eui';
+import React, { Fragment, useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { APP_PATH, BREADCRUMBS, PLUGIN_NAME } from '../../../utils/constants';
+import { SAMPLE_TYPE } from '../../../../server/utils/constants';
+import {
+  GET_SAMPLE_DETECTORS_QUERY_PARAMS,
+  GET_SAMPLE_INDICES_QUERY,
+} from '../../utils/constants';
+import { AppState } from '../../../redux/reducers';
+import { getDetectorList } from '../../../redux/reducers/ad';
+import { createSampleData } from '../../../redux/reducers/sampleData';
+
+import { getIndices, createIndex } from '../../../redux/reducers/opensearch';
+import { createDetector, startDetector } from '../../../redux/reducers/ad';
+import {
+  sampleHttpResponses,
+  sampleEcommerce,
+  sampleHostHealth,
+} from '../../SampleData/utils/constants';
+import {
+  containsSampleIndex,
+  containsSampleDetector,
+  getDetectorId,
+} from '../../SampleData/utils/helpers';
+import { SampleDataBox } from '../../SampleData/components/SampleDataBox/SampleDataBox';
+import { SampleDetailsFlyout } from '../../SampleData/components/SampleDetailsFlyout/SampleDetailsFlyout';
+import { prettifyErrorMessage } from '../../../../server/utils/helpers';
+import { CoreStart } from '../../../../../../src/core/public';
+import { CoreServicesContext } from '../../../components/CoreServices/CoreServices';
+import ContentPanel from '../../../components/ContentPanel/ContentPanel';
+import { CreateWorkflowStepDetails } from '../components/CreateWorkflowStepDetails';
+import { CreateWorkflowStepSeparator } from '../components/CreateWorkflowStepSeparator';
+
+interface AnomalyDetectionOverviewProps {
+  isLoadingDetectors: boolean;
+}
+
+export function AnomalyDetectionOverview(props: AnomalyDetectionOverviewProps) {
+  const core = React.useContext(CoreServicesContext) as CoreStart;
+  const dispatch = useDispatch();
+  const visibleIndices = useSelector(
+    (state: AppState) => state.opensearch.indices
+  );
+
+  const allDetectors = Object.values(
+    useSelector((state: AppState) => state.ad.detectorList)
+  );
+
+  const [isLoadingHttpData, setIsLoadingHttpData] = useState<boolean>(false);
+  const [isLoadingEcommerceData, setIsLoadingEcommerceData] = useState<boolean>(
+    false
+  );
+  const [isLoadingHostHealthData, setIsLoadingHostHealthData] = useState<
+    boolean
+  >(false);
+  const [
+    showHttpResponseDetailsFlyout,
+    setShowHttpResponseDetailsFlyout,
+  ] = useState<boolean>(false);
+  const [showEcommerceDetailsFlyout, setShowEcommerceDetailsFlyout] = useState<
+    boolean
+  >(false);
+  const [
+    showHostHealthDetailsFlyout,
+    setShowHostHealthDetailsFlyout,
+  ] = useState<boolean>(false);
+
+  const getAllSampleDetectors = async () => {
+    await dispatch(getDetectorList(GET_SAMPLE_DETECTORS_QUERY_PARAMS)).catch(
+      (error: any) => {
+        console.error('Error getting all detectors: ', error);
```

Comment:
I'm ok with that, will change. Makes sense here, since the returned error may not be cx-understandable.

Reply:
It's a fair question - I've seen this as common practice from this + other plugins to do something similar. I think it's ok as it's the best way for debugging reported custom issues like you mentioned. And because it's only visible in dev console.